src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1517,23 +1588,45 @@ MulticopterPositionControl::task_main()
 				}
 
 			} else if (_control_mode.flag_control_manual_enabled
-				   && _vehicle_land_detected.landed) {
-				/* don't run controller when landed */
-				_reset_pos_sp = true;
-				_reset_alt_sp = true;
-				_do_reset_alt_pos_flag = true;
-				_mode_auto = false;
-				reset_int_z = true;
-				reset_int_xy = true;
+				   && (landed_for_sure || (_landed_locked && _acc_z_lp < 4.0f))) {
```

Comment:
@Stifael The reason for this was to protect against free fall. Otherwise once the landed_locked flag is set in air falsely you have no chance of coming out again.
However, the real problem is that the accel value is not updated anymore once the the landed_locked flag is set. We need to fix that.